packages/catalog-model/src/kinds/util.ts

```diff
@@ -14,66 +14,18 @@
  * limitations under the License.
  */
 
-import Ajv, { AnySchema } from 'ajv';
-import * as yup from 'yup';
+import { entityKindSchemaValidator } from '../validation';
 import { KindValidator } from './types';
 
-/**
- * @deprecated We no longer use yup for the catalog model. This utility method will be removed.
- */
-export function schemaValidator(
-  kind: string,
-  apiVersion: readonly string[],
-  schema: yup.Schema<any>,
-): KindValidator {
-  return {
-    async check(envelope) {
-      if (kind !== envelope.kind || !apiVersion.includes(envelope.apiVersion)) {
-        return false;
-      }
-      await schema.validate(envelope, { strict: true });
-      return true;
-    },
-  };
-}
-
-export function ajvCompiledJsonSchemaValidator(
-  kind: string,
-  apiVersion: readonly string[],
-  schema: AnySchema,
-  extraSchemas?: AnySchema[],
-): KindValidator {
-  const ajv = new Ajv({ allowUnionTypes: true });
-  if (extraSchemas) {
-    ajv.addSchema(extraSchemas, undefined, undefined, true);
-  }
-  const validate = ajv.compile(schema);
-
+// TODO(freben): Left here as a compatibility helper. It would be nicer to
+// just export the inner validator directly. However, all of the already
+// exported kind validators have the `KindValidator` signature which is
+// different. So let's postpone that change until a later time.
```

Comment:
Should we deprecate it?

Reply:
This helper in and of itself is not exported, so it's fine to leave as is I think. The comment pertains to the specific kind validators that are the result of calling this one.